steamtinkerlaunch

```diff
@@ -7427,43 +7432,59 @@ function installd3d47dll {
 	fi
 }
 
+function createDLReShadeList {
+	RESHADEVERSIONS="$RSVERSOVRD!$RSVERS!5.0.0!4.9.1!3.4.1" # List some hardcoded reshade versions to selct for the override 
+	#TODO replace with a auto poplating list based on the last few releses
+}
+
 function dlReShade {
 	if [ -z "$1" ]; then
 		DLVERS="$RSVERS"
 	else
 		DLVERS="$1"
 	fi
-	rm "$RESHADESRCDIR/$RS_64" 2>/dev/null
-	rm "$RESHADESRCDIR/$RS_32" 2>/dev/null
-	rm "$RESHADESRCDIR/${RS_64//.dll/.json}" 2>/dev/null
-	rm "$RESHADESRCDIR/${RS_32//.dll/.json}" 2>/dev/null
 
-	DLDST="${RESHADESRCDIR}/${RSSU}.exe"
-	DLDSTV="${RESHADESRCDIR}/${RSSU}_${DLVERS}.exe"
+	# No longer needed?
+	#rm "$RESHADESRCDIR/$RS_64" 2>/dev/null
+	#rm "$RESHADESRCDIR/$RS_32" 2>/dev/null
+	#rm "$RESHADESRCDIR/${RS_64//.dll/.json}" 2>/dev/null
+	#rm "$RESHADESRCDIR/${RS_32//.dll/.json}" 2>/dev/null
+
+	#DLDST="${RESHADESRCDIR}/${RSSU}.exe"
+	#DLDSTV="${RESHADESRCDIR}/${RSSU}_${DLVERS}.exe"
+
+	DLDST="${RESHADESRCDIR}/${RSSU}_${DLVERS}.exe"
 
 	RSSETUP="${RESHADEDLURL}/${RSSU}_${DLVERS}.exe"
 
 	dlCheck "$RSSETUP" "$DLDST" "X" "Downloading $RSSU"
-	if [ ! -f "$DLDSTV" ]; then
-		writelog "INFO" "${FUNCNAME[0]} - Creating backup '${RSSU}_${DLVERS}.exe'"
-		cp "$DLDST" "$DLDSTV"
-	fi
-
+	
 	echo "$DLVERS" > "${DLDST//.exe/.log}"
 
 	if [ ! -s "$DLDST" ]; then
 		writelog "SKIP" "${FUNCNAME[0]} - Downloaded file '$DLDST' is empty - removing"
 		rm "$DLDST" 2>/dev/null
 	else
-		"$UNZIP" -q "$DLDST" -d "$RESHADESRCDIR" 2>/dev/null
+		"$UNZIP" -q "$DLDST" -d "$RESHADESRCDIR/$RSVERS" 2>/dev/null
 		writelog "INFO" "${FUNCNAME[0]} - Downloaded and extracted ${RESH}-v${DLVERS} file '$DLDST'"
 	fi
 }
 
+function overrideReShadeVersion
+{
+	if ! [ "$RSVERSOVRD" -eq "$RSVERS" ]; then
+		writelog "INFO" ${FUNCNAME[0]} - "Overriding ReShade version - Set to '$RSVERSOVRD'"
+		RSVERS="$RSVERSOVRD"
+	else
+		writelog "SKIP" ${FUNCNAME[0]} - "RSVERSOVRD is set to '$RSVERSOVRD' continue with reshade version - '$RSVERS'"
+	fi
+}
+
 # prepare reshade files if not found:
 function prepareReshadeFiles {
-	if [ "$DOWNLOAD_RESHADE" -eq 1 ]; then
-		writelog "INFO" "${FUNCNAME[0]} - DOWNLOAD_RESHADE enabled"
+overrideReShadeVersion
+if [ "$DOWNLOAD_RESHADE" -eq 1 ]; then
+	writelog "INFO" "${FUNCNAME[0]} - DOWNLOAD_RESHADE enabled"
```

Comment:
Really nitpicky and not sure if GitHub is just messing up the formatting but does this need indented?

Reply:
huh shouldn't be indenting that much it should just be a small indent to show it goes in the if ill take a look at that

Reply:
On GitHub it's showing up as not indented at all on those lines but it could be a GitHub glitch :-)

![image](https://user-images.githubusercontent.com/7917345/200982852-c36be4be-5d25-4ede-99d0-85d0514c9e34.png)

Reply:
Thanks, resolved now :-)